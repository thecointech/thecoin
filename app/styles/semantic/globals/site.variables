/*******************************
     User Global Variables
*******************************/

/*-------------------
       Fonts
--------------------*/

@fontName          : 'Muli';

/*-------------------
    Brand Colors
--------------------*/

@primaryColor        : #415DB0;
@secondaryColor      : #3193A1;

@complementaryColor	 : #FFC64A

@lightPrimaryColor   : #617AC5;
@lightSecondaryColor : #51ADBA;

@lightComplementaryColor: #FFDE98

/*-------------------
        Page
--------------------*/

@pageBackground      : #F0F0F0;

@lineHeight          : 1.6em;
@textColor           : rgba(0, 0, 0, 0.87);

