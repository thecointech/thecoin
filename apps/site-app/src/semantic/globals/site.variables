/*******************************
     Site-specific Global Variables
*******************************/

@pageBackground      : @theCoinPaletteGray2;
