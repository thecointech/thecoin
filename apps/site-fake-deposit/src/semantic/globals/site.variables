/*******************************
     Site-specific Global Variables
*******************************/

//@pageBackground      : @green;
