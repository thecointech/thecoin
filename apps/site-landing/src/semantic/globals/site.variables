/*******************************
     Site-specific Global Variables
*******************************/

//@pageBackground      : @green;

// A common definition for columns because we can't use a single grid for
// the entire page, so this provides single source for columns
@pageGridLeft: .scalable(20, 131)[@result];
@pageGridCenter: minmax(auto, 1178px);
@pageGridRight: .scalable(20, 131)[@result];
@pageGridContent: @pageGridLeft @pageGridCenter @pageGridRight;
