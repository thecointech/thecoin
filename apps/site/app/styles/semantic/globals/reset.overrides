/*******************************
         Site Overrides
*******************************/
*{
	margin: 0;
	padding: 0;
  border: none;
  box-shadow: none;
	font-size: 100%;
	font: inherit;
	vertical-align: baseline;
  background: transparent;
  color: inherit;
  padding: 0;
}

.ui.segments .segment, .ui.container .ui.segment, .ui.container{
  border: none;
  box-shadow: none;
  margin-top: 10px;
}

.ui.container {
  margin-left: 0 !important;
  margin-right: 0 !important;
}