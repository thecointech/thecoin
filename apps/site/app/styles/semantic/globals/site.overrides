/*******************************
         Site Overrides
*******************************/

a{
    font-size: 12px;
    font-weight: @bold;
    text-transform: uppercase;
    padding-bottom: 10px;
    border-bottom: 2px solid @linkUnderline;
}