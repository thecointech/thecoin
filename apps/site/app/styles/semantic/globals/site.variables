/*******************************
     User Global Variables
*******************************/

/*-------------------
        Page
--------------------*/

@pageBackground      : @white;
@maxContentWidth: 1440px;

/*-------------------
       Fonts
--------------------*/

@fontName2         : 'Abril Fatface';
@fontName          : 'Montserrat';

@headerFont        : @fontName2, 'Helvetica Neue', Arial, Helvetica, sans-serif;
@pageFont          : @fontName, 'Helvetica Neue', Arial, Helvetica, sans-serif;


/*-------------------
    Brand Colors
--------------------*/

@primaryColor        : #F18182;
@secondaryColor      : #20B58D;

/*--------------
  Page Heading
---------------*/

@headerFontWeight : @bold;
@headerLineHeight : unit((18 / 14), em);

@h1 : unit((80 / 14), rem);
@h2 : unit((64 / 14), rem);
@h3 : unit((40 / 14), rem);
@h4 : unit((28 / 14), rem);
@h5 : unit((14 / 14), rem);


/*-------------------
       Links
--------------------*/

@linkColor           : #61C1B8;
@linkUnderline       : @primaryColor;
@linkHoverColor      : darken(saturate(@linkColor, 20), 15, relative);
@linkHoverUnderline  : @linkUnderline;


/*-------------------
     Transitions
--------------------*/

@defaultDuration : 0.1s;
@defaultEasing   : ease;


/*-------------------
     Screen sizes
--------------------*/

@small: ~"(max-width: 699px) and (min-width: 520px)";
@medium: ~"(max-width: 1000px) and (min-width: 700px)";
@large: ~"(min-width: 1001px)";
@xlarge: ~"(min-width: 1151px)";