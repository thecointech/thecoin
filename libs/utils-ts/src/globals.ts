import { Firestore } from '@thecointech/types';

type TheCoinGlobals = {
  firestore: Firestore|null;
}
declare global {
  namespace globalThis {
    //  deepcode ignore no-var-keyword: var is necessary for this typing to work
    var __thecoin: TheCoinGlobals
  }
}
// initialize to empty
globalThis.__thecoin = {
  firestore: null,
}
