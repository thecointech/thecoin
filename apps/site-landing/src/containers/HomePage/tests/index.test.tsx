/**
 * Test the HomePage
 */

import React from 'react';
import { jest } from '@jest/globals';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { HomePage } from '../index';
import { MemoryRouter } from 'react-router';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(() => ({})),
  // useSelector: jest.fn(selector => selector({
  //   accounts: {
  //     map: {},
  //     active: undefined,
  //   }
  // })),
}));

describe('<HomePage />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      // tslint:disable-next-line: jsx-wrap-multiline
      <IntlProvider locale="en">
        <MemoryRouter>
          <HomePage />
        </MemoryRouter>
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
