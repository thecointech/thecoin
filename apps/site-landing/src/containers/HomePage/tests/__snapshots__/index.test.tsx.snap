// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div>
  <div />
  <div>
    <div>
      <h1
        class="ui header"
      >
        The future is brighter
        <div
          class="sub header"
        >
          <span>
            Save, invest and spend money with TheCoin,
          </span>
          <span>
            get 100% of benefits and save our Planet.
          </span>
        </div>
      </h1>
      <a
        class="ui red large button"
        href="http://localhost:3001"
        role="button"
      >
        Start Now
      </a>
    </div>
    <div>
      <div>
        <h4
          class="ui header"
        >
          You’re Wealthier
        </h4>
        <p>
          Your money is always earning. Make it earn for you.
        </p>
        <a
          href="/compare"
        >
          Learn More
          <div
            class="undefined x2spaceBefore"
          />
        </a>
      </div>
      <div>
        <h4
          class="ui header"
        >
          Earth’s Healthier
        </h4>
        <p>
          Be part of the solution. We offset our clients’ CO2.
        </p>
        <a
          href="/wedomore"
        >
          Learn More
          <div
            class="undefined x2spaceBefore"
          />
        </a>
      </div>
    </div>
  </div>
  <div>
    <div
      class="ui stackable center aligned middle aligned equal width grid"
    >
      <div
        class="row"
      >
        <div
          class="two wide column"
        >
          <h2>
            0
          </h2>
          <h4>
            %
          </h4>
          <br />
          Monthly Fees
        </div>
        <div
          class="computer only tablet only one wide column"
        >
          <hr />
        </div>
        <div
          class="two wide column"
        >
          <h2>
            0
          </h2>
          <h4>
            %
          </h4>
          <br />
          Minimums
        </div>
        <div
          class="computer only tablet only one wide column"
        >
          <hr />
        </div>
        <div
          class="three wide column"
        >
          <h2>
            100
          </h2>
          <h4>
            %
          </h4>
          <br />
          Your benefits
        </div>
      </div>
    </div>
  </div>
  <div />
  <div>
    <div>
      <h2
        class="ui header"
      >
        You’re Wealthier
        <div
          class="sub header x5spaceBefore"
        >
          Enjoy the benefits of investment growth on every dollar with our online spending account.
        </div>
      </h2>
    </div>
    <div>
      <div>
        <img
          src="[object Object]"
        />
        <h4
          class="ui header"
        >
          Start Now
        </h4>
        <p>
          The most important ingredient is time. You’re young - maximize your benefit!
        </p>
      </div>
      <div>
        <a
          href="/blog/start-now"
        >
          Learn Your Edge
          <div
            class="undefined x2spaceBefore"
          />
        </a>
      </div>
    </div>
    <div>
      <div>
        <img
          src="[object Object]"
        />
        <h4
          class="ui header"
        >
          The Best Investment
        </h4>
        <p>
          Why settle? Our time-tested strategy has the best long-term results.
        </p>
      </div>
      <div>
        <a
          href="/compare"
        >
          Compare Us
          <div
            class="undefined x2spaceBefore"
          />
        </a>
      </div>
    </div>
    <div>
      <img
        src="[object Object]"
      />
    </div>
    <div>
      <div>
        <img
          src="[object Object]"
        />
        <h4
          class="ui header"
        >
          Maximum Growth
        </h4>
        <p>
          Add $250000 to your retirement - with no money down.
        </p>
      </div>
      <div>
        <a
          href="/compare"
        >
          How Anyone Can Get Rich Slow
          <div
            class="undefined x2spaceBefore"
          />
        </a>
      </div>
    </div>
    <div>
      <div>
        <img
          src="[object Object]"
        />
        <h4
          class="ui header"
        >
          Keep Your Profits
        </h4>
        <p>
          Every dollar earns every day. As a non-profit, we want you to keep that benefit.
        </p>
      </div>
      <div>
        <a
          href="/healthier"
        >
          Who Wins?
          <div
            class="undefined x2spaceBefore"
          />
        </a>
      </div>
    </div>
  </div>
  <div>
    <h3
      class="ui header"
    >
      TheCoin is a revolutionary new kind of account.
    </h3>
    <div>
      <a
        class="ui red large button"
        href="http://localhost:3001/#/addAccount"
        role="button"
      >
        Create Account
      </a>
    </div>
  </div>
  <div>
    <div>
      <img
        src="[object Object]"
      />
    </div>
    <div>
      <h2
        class="ui header"
      >
        Earths’ Healthier
        <div
          class="sub header x5spaceBefore"
        >
          <span>
            You can become CO2-neutral with TheCoin
          </span>
          <span>
            as we work to offset our clients’ emissions.
          </span>
        </div>
      </h2>
      <div>
        <div>
          <div>
            <img
              src="[object Object]"
            />
            <h4
              class="ui header"
            >
              Make a Difference
            </h4>
            <p>
              Offsetting CO2 is effective. Extremely effective!
            </p>
          </div>
          <div>
            <a
              href="/healthier"
            >
              Compare Outcomes
              <div
                class="undefined x2spaceBefore"
              />
            </a>
          </div>
        </div>
        <div>
          <div>
            <img
              src="[object Object]"
            />
            <h4
              class="ui header"
            >
              Scientifically Verified
            </h4>
            <p>
              Prove it? We’d love to!
            </p>
          </div>
          <div>
            <a
              href="/healthier"
            >
              Why We Can Be So Confident
              <div
                class="undefined x2spaceBefore"
              />
            </a>
          </div>
        </div>
        <div>
          <div>
            <img
              src="[object Object]"
            />
            <h4
              class="ui header"
            >
              We Do More
            </h4>
            <p>
              It’s not just CO2. The projects we fund are vital to restoring our ecosystem.
            </p>
          </div>
          <div>
            <a
              href="/healthier"
            >
              What We Do
              <div
                class="undefined x2spaceBefore"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="undefined undefined "
  >
    <h3
      class="ui header"
    >
      The benefits of a chequing, savings, and investing account all in one!
    </h3>
    <a
      class="ui red large button"
      href="http://localhost:3001/#/addAccount"
      role="button"
    >
      Create Account
    </a>
  </div>
</div>
`;
